ReactWindows/ReactNative.Net46/Views/TextInput/ReactPasswordBoxManager.cs

```diff
@@ -0,0 +1,458 @@
+        public override void OnDropViewInstance(ThemedReactContext reactContext, PasswordBox view)
+        {
+            base.OnDropViewInstance(reactContext, view);
+            view.KeyDown -= OnKeyDown;
+            view.LostFocus -= OnLostFocus;
+            view.GotFocus -= OnGotFocus;
+            view.PasswordChanged -= OnPasswordChanged;
+        }
+
+        /// <summary>
+        /// Sets the dimensions of the view.
+        /// </summary>
+        /// <param name="view">The view.</param>
+        /// <param name="dimensions">The output buffer.</param>
+        public override void SetDimensions(PasswordBox view, Dimensions dimensions)
+        {
+            Canvas.SetLeft(view, dimensions.X);
+            Canvas.SetTop(view, dimensions.Y);
+            view.Width = dimensions.Width;
+        }
+
+#endregion
+
+#region Events
+
+        private void OnPasswordChanged(object sender, RoutedEventArgs e)
+        {
+            var textBox = (PasswordBox)sender;
+            textBox.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(
+                    new ReactTextChangedEvent(
+                        textBox.GetTag(),
+                        textBox.Password,
+                        textBox.ActualWidth,
+                        textBox.ActualHeight,
+                        0));
+        }
+
+        private void OnGotFocus(object sender, RoutedEventArgs e)
+        {
+            var textBox = (PasswordBox)sender;
+            textBox.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher
+                .DispatchEvent(
+                    new ReactTextInputFocusEvent(textBox.GetTag()));
+        }
+
+        private void OnLostFocus(object sender, RoutedEventArgs e)
+        {
+            var textBox = (PasswordBox)sender;
+            var eventDispatcher = textBox.GetReactContext()
+                .GetNativeModule<UIManagerModule>()
+                .EventDispatcher;
+
+            eventDispatcher.DispatchEvent(
+                new ReactTextInputBlurEvent(textBox.GetTag()));
+
+            eventDispatcher.DispatchEvent(
+                new ReactTextInputEndEditingEvent(
+                      textBox.GetTag(),
+                      textBox.Password));
+        }
+        
+        private void OnKeyDown(object sender, KeyEventArgs e)
+        {
+            if (e.Key == Key.Enter)
+            {
+                var textBox = (PasswordBox)sender;
+                e.Handled = true;
+                textBox.GetReactContext()
+                    .GetNativeModule<UIManagerModule>()
+                    .EventDispatcher
+                    .DispatchEvent(
+                        new ReactTextInputSubmitEditingEvent(
+                            textBox.GetTag(),
+                            textBox.Password));
+            }
+        }
+
+#endregion
```

Comment:
If we're splitting the code, let's ditch the 2003-era #region statements while we're at it.